includes/woopay/services/class-checkout-service.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class Create_And_Confirm_Intention_Test
+ *
+ * @package Checkout_Service
+ */
+
+namespace WCPay\WooPay\Service;
+
+use WC_Payments_Features;
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WCPay\Core\Server\Request\WooPay_Create_And_Confirm_Intention;
+
+/**
+ * Checkout service class.
+ */
+class Checkout_Service {
+
+	/**
+	 * Create woopay request from base create and confirm request.
+	 *
+	 * @param Request   $base_request Base request.
+	 * @param \WC_Order $order Order.
+	 * @param bool      $using_saved_payment_method Using saved payment method.
+	 *
+	 * @return WooPay_Create_And_Confirm_Intention
+	 * @throws Invalid_Request_Parameter_Exception
+	 * @throws \WCPay\Core\Exceptions\Extend_Request_Exception
+	 */
+	public function create_woopay_intention_request( $base_request, $order, $using_saved_payment_method ) {
+		if ( ! $order ) {
```

Comment:
As we control the code within the gateway, I'd trust it not to changes those filters. As a solution in the middle, let's define types for the parameters of the method, and PHP will deal with it.

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/5136/commits/8f1aa8fd35a4f80de54b317d8892d0aae95753cd . The payment information object is passed that holds the order object.